### PR TITLE
middleware added in specific routes and avoided for specific and rese…

### PR DESCRIPTION
**Pull request by aritrikaTuteck**
middleware added in specific routes and avoided for specific and reset_password completed

### CODE REVIEW

File: middleware/auth.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from fastapi import Request;
+from starlette.middleware.base import BaseHTTPMiddleware;
+from jose import JWTError, jwt;
+import os;
+
+class auth_middleware(BaseHTTPMiddleware):
+  async def dispatch(self, request : Request, call_next):
+    #print(request.headers.get("authorization"));
+    if request.url.path == "/authorization_authentication/password/reset":
```
Comment (mehul-tuteck) [inline middleware/auth.py:9]: This URL path does not follow REST API standards.

File: middleware/auth.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      authorized_token:str = request.headers.get("authorization");
+      token = authorized_token.split(" ")[1];
+      decoded_jwt = jwt.decode(token, os.getenv("JWT_SECRET_KEY"), algorithms= os.getenv("JWT_ALGORITHM"));
+      request.state.user_id = decoded_jwt.get("userId");
+      response = await call_next(request)
+     # print(request)
```
Comment (mehul-tuteck) [inline middleware/auth.py:15]: Let's remove unnecessary comments.

File: routers/auth.py [hunk 3]
```diff
@@ -1,15 +1,21 @@
-from fastapi import APIRouter,Response
+from fastapi import APIRouter, Response, Header;
 from email_validator import validate_email,EmailNotValidError
```
Comment (mehul-tuteck) [inline routers/auth.py:2]: Please keep the following convention in mind to increase code readability 

`import third party`
space
`import local`

File: routers/auth.py [hunk 4]
```diff
@@ -97,10 +100,23 @@ async def forgot_password():
...
-        return SuccessResponse(data=None,client_msg="You are successfully registered!",dev_msg="Registration Successful!")
+        user_id = request.state.user_id;
+        request_body_data = await request.json();
+        new_password = request_body_data.get("new_password");
+       
+        if user_id and new_password:
+            user = fetch_user_with_id(user_id);
+            if user:
+                set_new_password(user_id, new_password);
+            else:
```
Comment (mehul-tuteck) [inline routers/auth.py:114]: Please fix the client messages

File: routers/auth.py [hunk 5]
```diff
@@ -97,10 +100,23 @@ async def forgot_password():
         return ServerError(err=e,errMsg=str(e))
     
 
-@router.post("/password/reset")
-async def reset_password():
+@router.patch("/password/reset")
+async def reset_password(request: Request):
```
Comment (mehul-tuteck) [inline routers/auth.py:104]: Lets name all incoming request bodies as request, but their types should be understandable

It should be `request : ResetPassword`

The types should be very clear

**Review by mehul-tuteck — CHANGES_REQUESTED** [review]

File: middleware/auth.py (hunk 2 above)
Comment (aritrikaTuteck) [reply middleware/auth.py:15]: Done

**Pull request merged**